server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/utils/SequentialDocValuesIterator.java

```diff
@@ -47,6 +47,49 @@ public SequentialDocValuesIterator(DocIdSetIterator docIdSetIterator) {
         this.docIdSetIterator = docIdSetIterator;
     }
 
+    /**
+     * Constructs a new SequentialDocValuesIterator instance with the given SortedNumericDocValues.
+     *
+     */
+    public SequentialDocValuesIterator() {
```

Comment:
If the intention is to have empty sorted numeric doc values.
Then we can use in the callers :

```
new SequentialDocValuesIterator(DocValues.emptySortedNumeric());
```

Reply:
do we really need this constructor ? wherever we want empty sorted numeric values, we can initialize that way.

Now with this change, sequential doc values iterator is tied to sorted numeric - which makes it not generic.